packages/scandipwa/src/component/MyAccountMyWishlist/MyAccountMyWishlist.component.js

```diff
@@ -129,6 +129,49 @@ export class MyAccountMyWishlist extends PureComponent {
 
     renderProduct([id, product]) {
         const { isEditingActive, loadingItemsMap, setIsQtyUpdateInProgress } = this.props;
+        const {
+            wishlist = {}
+        } = product;
+
+        // assign wishlist.options to a new list
+        // remove the first part of the value string that indicated quantity
+        // append with a value from buy_request instead
+        // get the value using index
+        if (wishlist.options.length >= 1) {
+            // the order of items in wishlist.options always corresponds with the order of items
+            // in buy_request.
+            // we're using regex instead of JSON parse bc we need to preserve the order of the data
+            // objects are inherently unordered
+            const { buy_request } = wishlist;
+            const selections = buy_request.match(/bundle_option(.*)bundle/g)[0].match(/\d+/g);
```

Comment:
If the regex fails, the `match` function will return null and this code will crash.

Reply:
for all bundled products the buy_request follows the same convention, regex won't fail unless somebody drastically changes how the wishlist works on the backend, but then it'll be a different problem.

parsing JSON turns the string into an object, which are unordered. working around that would add computational load to the already overly complex component.